src/main/java/games/strategy/triplea/ui/AbstractMovePanel.java

```diff
@@ -252,7 +238,12 @@ protected final void display(final PlayerID id, final String actionLabel) {
       if (setCancelButton()) {
         add(leftBox(cancelMoveButton));
       }
-      add(leftBox(new JButton(doneMoveAction)));
+      add(leftBox(new JButton(SwingAction.of("Done", e -> {
```

Comment:
I just noticed that removing the WeakReference was a neccessity.
Before the field doneMove held another reference to the listener object which kept it in memory.
When removing the field this would have been no longer the case...